library/src/blas1/rocblas_asum_batched.hpp

```diff
@@ -0,0 +1,56 @@
+/* ************************************************************************
```

Comment:
> It might be worthwhile to combine all of the .hpp files into a single .hpp file, if we are thinking about combining the templates in the future anyway.

That is a good idea, as long as the separate templates are somewhat related, and could later be refactored into simpler code (such as wrappers to a more general common template).

If a template in a `.hpp` file is not instantiated (used), it will not affect program output, just maybe compile time. So there's no problem with having multiple templates, some of which are unused, in the same header file.